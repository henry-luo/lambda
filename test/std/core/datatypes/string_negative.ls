// Test: String Negative Cases
// Category: core/datatypes
// Type: negative
// Expected: error

// Invalid string operations that should fail
"hello" + 42
"world" - "test"
"abc" * "def"
"hello" / 2

// Invalid string indexing
let str = "hello"
str[10]  // Index out of bounds
str[-1]  // Negative index

// Invalid method calls
str.invalidMethod()
