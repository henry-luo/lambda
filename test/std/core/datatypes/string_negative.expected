09:15:42 [NOTICE] Start transpiling test/std/core/datatypes/string_negative.ls...
[script:list]
  [content:list[8]]
    no declare
    item:
    [binary expr +.3:any]
      [primary expr:string,const:1]
        [const@0, string, 0x14b80aa30, 'hello']
      [primary expr:int,const:1]
        [const: int]
    item:
    [binary expr -.4:any]
      [primary expr:string,const:1]
        [const@1, string, 0x14b80ab98, 'world']
      [primary expr:string,const:1]
        [const@2, string, 0x14b80ac28, 'test']
    item:
    [binary expr *.5:any]
      [primary expr:string,const:1]
        [const@3, string, 0x14b80ad40, 'abc']
      [primary expr:string,const:1]
        [const@4, string, 0x14b80adc8, 'def']
    item:
    [binary expr /.7:any]
      [primary expr:string,const:1]
        [const@5, string, 0x14b80aed8, 'hello']
      [primary expr:int,const:1]
        [const: int]
    item:
    [let stam:null]
      declare:
      [assign expr:str:string]
        [primary expr:string,const:1]
          [const@6, string, 0x14b80b090, 'hello']
    item:
    [primary expr:any,const:0]
      [index expr:any]
        object:
        [primary expr:string,const:1]
          [ident:str:string,const:1]
        field:
        [primary expr:int,const:1]
          [const: int]
    item:
    [primary expr:any,const:0]
      [index expr:any]
        object:
        [primary expr:string,const:1]
          [ident:str:string,const:1]
        field:
        [primary expr:int,const:1]
          [const: int]
    item:
    [primary expr:any,const:0]
      [call expr:any]
        [primary expr:any,const:0]
          [member expr:any]
            object:
            [primary expr:string,const:1]
              [ident:str:string,const:1]
            field:
            [ident:invalidMethod:any,const:0]
        args:
09:15:42 [NOTICE] compiling C code in 'test/std/core/datatypes/string_negative.ls' to MIR
C2MIR init end           -- 69 usec
test/std/core/datatypes/string_negative.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2890, parse tokens -- 1832
  C2MIR preprocessor end    -- 544 usec
  C2MIR parser end          -- 671 usec
test/std/core/datatypes/string_negative.ls:452:16: warning -- using pointer without cast for integer type parameter
test/std/core/datatypes/string_negative.ls:453:16: warning -- using pointer without cast for integer type parameter
  C2MIR context checker end -- 764 usec
  C2MIR generator end       -- 832 usec
C2MIR compiler end                -- 867 usec
09:15:42 [NOTICE] Generating native code...
09:15:42 [ERROR]   unknown add type: 10, 3
09:15:42 [ERROR]   unknown sub type: 10, 10

09:15:42 [ERROR]   unknown mul type: 10, 10

09:15:42 [ERROR]   unknown div type: 10, 3

##### Script 'test/std/core/datatypes/string_negative.ls' executed: #####
error
error
error
error
